File: Algorithms Data Structures/permutationString/permutationString.cpp
#include <iostream>
#include <string>

using namespace std;

bool checkPermutation(char *&str1, char *&str2, int length){
    int letters[128] {};
    for (int i = 0; i < length; i++){
        int c = int(*(str1 + i));
        letters[c]++;
    }
    for (int i = 0; i < length; i++){
        int c = int(*(str2 + i));
        letters[c]--;
        if (letters[c] < 0){
            return false;
        }
    }
    return true;
}
int main(int argc, char const *argv[])
{
    cout << "Enter two strings for checking: " << endl;
    const char * str1;
    str1 = "Hello";
    cout << *(str1+1) << endl;
    /*
    char *str1;
    char *str2;
    string inputStr;
    getline(cin, inputStr);
    string delimiter = " ";
    string token;
    int pos = 0;
    while((pos = inputStr.find(delimiter)) != string::npos){
        token = inputStr.substr(0, pos);
        str1 = token;
        inputStr.erase(0, pos + delimiter.length());
    }
    */
    return 0;
}

File: Algorithms Data Structures/permutationString/permutationString.java
import java.util.Arrays;
import java.util.Scanner;

/**
 * permutationString
 */
public class permutationString {

    static boolean checker(String str1, String str2){
        if (str1.length() != str2.length()){
            return false;
        }
        int[] letters = new int[128];
        char[] c1 = str1.toCharArray();
        char[] c2 = str2.toCharArray();

        for (int i = 0; i < c1.length; i++){
            int val = c1[i];
            letters[i]++;
        }

        for (int j = 0; j < c2.length; j++){
            val = c2[i];
            letters[val]--;
        }

        for (int i = 0; i < letters.length; i++){
            if (letters[i] != 0){
                return false;
            }
        }
        return true;
    }
    /*static boolean checker(String str1, String str2){
        int[] letters = new int[128];
        char[] c1 = str1.toCharArray();
        char[] c2 = str2.toCharArray();
        if (str1.length() != str2.length()){
            return false;
        }
        for (int i = 0; i < str1.length(); i++){
            int val = (int)str1.charAt(i);
            letters[val]++;
        }
        for (int i = 0; i < str1.length(); i++){
            int c = (int)str2.charAt(i);
            letters[c]--;
            if (letters[c] < 0){
                return false;
            }
        }
        return true;
    }
    */
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.print("Enter two strings for checking: ");
        String initialStr = in.nextLine();
        String[] strArr = initialStr.split(" ");
        System.out.println(checker(strArr[0], strArr[1]));
    }
}